api/src/test/java/com/epam/pipeline/acl/metadata/MetadataApiServiceTest.java

```diff
@@ -0,0 +1,453 @@
+        initAclEntity(anotherEntity);
+        doReturn(mutableListOf(entry, anotherEntry)).when(mockMetadataManager).loadEntitiesMetadataFromFolder(ID);
+        mockLoadEntity(entity, ID);
+        mockLoadEntity(anotherEntity, ID_2);
+        mockAuthentication(SIMPLE_USER);
+
+        assertThat(metadataApiService.loadEntitiesMetadataFromFolder(ID)).hasSize(1).contains(entry);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldDenyLoadEntitiesMetadataFromFolderWhenPermissionIsNotGranted() {
+        initAclEntity(anotherEntity);
+        doReturn(mutableListOf(anotherEntry)).when(mockMetadataManager).loadEntitiesMetadataFromFolder(ID);
+        mockLoadEntity(anotherEntity, ID_2);
+        mockAuthentication(SIMPLE_USER);
+
+        assertThat(metadataApiService.loadEntitiesMetadataFromFolder(ID)).isEmpty();
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldSearchMetadataByClassAndKeyValueForAdmin() {
+        doReturn(mutableListOf(entityVO)).when(mockMetadataManager)
+                .searchMetadataByClassAndKeyValue(ENTITY_ACL_CLASS, TEST_STRING, TEST_STRING);
+
+        assertThat(metadataApiService.searchMetadataByClassAndKeyValue(ENTITY_ACL_CLASS, TEST_STRING, TEST_STRING))
+                .hasSize(1).contains(entityVO);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldSearchMetadataByClassAndKeyValueWhenPermissionIsGranted() {
+        initAclEntity(entity, AclPermission.READ);
+        doReturn(mutableListOf(entityVO)).when(mockMetadataManager)
+                .searchMetadataByClassAndKeyValue(ENTITY_ACL_CLASS, TEST_STRING, TEST_STRING);
+        mockLoadEntity(entity, ID);
+        mockAuthentication(SIMPLE_USER);
+
+        assertThat(metadataApiService.searchMetadataByClassAndKeyValue(ENTITY_ACL_CLASS, TEST_STRING, TEST_STRING))
+                .hasSize(1).contains(entityVO);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldSearchMetadataByClassAndKeyValueWhichPermissionIsGranted() {
+        initAclEntity(entity, AclPermission.READ);
+        initAclEntity(anotherEntity);
+        doReturn(mutableListOf(entityVO, anotherEntityVO)).when(mockMetadataManager)
+                .searchMetadataByClassAndKeyValue(ENTITY_ACL_CLASS, TEST_STRING, TEST_STRING);
+        mockLoadEntity(entity, ID);
+        mockLoadEntity(anotherEntity, ID_2);
+        mockAuthentication(SIMPLE_USER);
+
+        assertThat(metadataApiService.searchMetadataByClassAndKeyValue(ENTITY_ACL_CLASS, TEST_STRING, TEST_STRING))
+                .hasSize(1).contains(entityVO);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldDenySearchMetadataByClassAndKeyValueWhenPermissionIsNotGranted() {
+        initAclEntity(anotherEntity);
+        doReturn(mutableListOf(anotherEntityVO)).when(mockMetadataManager)
+                .searchMetadataByClassAndKeyValue(ENTITY_ACL_CLASS, TEST_STRING, TEST_STRING);
+        mockLoadEntity(anotherEntity, ID_2);
+        mockAuthentication(SIMPLE_USER);
```

Comment:
What do you think about moving `mockAuthentication` call to the test class setup method? User name can be retrieved from the security context. It will also require to change some of the method `@WithMockUser` annotations.

Reply:
Discussed and refused.